File: rust/src/models/usage_model_enum.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(
    Clone, Copy, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize, Deserialize,
)]
pub enum UsageModelEnum {
    #[default]
    #[serde(rename = "PER_UNIT")]
    PerUnit,

    #[serde(rename = "TIERED")]
    Tiered,

    #[serde(rename = "VOLUME")]
    Volume,

    #[serde(rename = "PACKAGE")]
    Package,

    #[serde(rename = "MATRIX")]
    Matrix,
}

impl fmt::Display for UsageModelEnum {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = match self {
            Self::PerUnit => "PER_UNIT",
            Self::Tiered => "TIERED",
            Self::Volume => "VOLUME",
            Self::Package => "PACKAGE",
            Self::Matrix => "MATRIX",
        };
        f.write_str(value)
    }
}

File: typescript/src/models/usageModelEnum.ts
// this file is @generated

export enum UsageModelEnum {
  PerUnit = "PER_UNIT",
  Tiered = "TIERED",
  Volume = "VOLUME",
  Package = "PACKAGE",
  Matrix = "MATRIX",
}

export const UsageModelEnumSerializer = {
  _fromJsonObject(object: any): UsageModelEnum {
    return object;
  },

  _toJsonObject(self: UsageModelEnum): any {
    return self;
  },
};
